### PR TITLE
Disable Spark 3.4.0 premerge for 22.08 and enable for 22.10 

### PR DESCRIPTION
**Pull request by tgravescs**
fixes https://github.com/NVIDIA/spark-rapids/issues/5808

Disable the Spark 3.4 build for premerge since its not stable and changing frequently.  We will re-enable in 22.10 when its closer to release.  
Signed-off-by: Thomas Graves <tgraves@nvidia.com>

### CODE REVIEW

**Comment by tgravescs** [comment]
build

**Review by revans2 — APPROVED** [review]

**Review by jlowe — DISMISSED** [review]

File: jenkins/spark-premerge-build.sh [hunk 1]
```diff
@@ -130,10 +130,18 @@ nvidia-smi
 
 . jenkins/version-def.sh
 
-BUILD_SNAPSHOTS="false"
+# controls whether we build snapshots for the Spark maintenance versions like 3.1.4 and 3.2.2
```
[the remark this answers is not in the document]
Comment (tgravescs) [reply jenkins/spark-premerge-build.sh:133]: updated.

**Comment by tgravescs** [comment]
build

**Comment by gerashegalov** [comment]
It's still trying to include 340 because it's part of snapshot.buildvers in dist/pom.xml but now it came from cache (we still need a single source of truth to avoid this) :
```
[2022-06-13T22:10:21.501Z] [ERROR] /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-4907/dist/maven-antrun/build-parallel-worlds.xml:163: exec check-shims-revisions.sh failed, exit code is 1, error msg is Check Failed: git revisions between shims are not equal
...
Error: 6-13T22:10:21.501Z] [ERROR] Version file /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-4907/dist/target/parallel-world/spark330/rapids4spark-version-info.properties, revision is 65f2d79e1cc6fda420ffe5a0716da54d9b58c1d0
Error: 6-13T22:10:21.501Z] [ERROR] Version file /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-4907/dist/target/parallel-world/spark340/rapids4spark-version-info.properties, revision is 165ec8127f829ede29c69cff729fcee82bcbd8e2
```

**Comment by pxLi** [comment]
> It's still trying to include 340 because it's part of snapshot.buildvers in dist/pom.xml but now it came from cache (we either need to update the property and file an issue for a single source of truth to avoid running into this) :
> 
> ```
> [2022-06-13T22:10:21.501Z] [ERROR] /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-4907/dist/maven-antrun/build-parallel-worlds.xml:163: exec check-shims-revisions.sh failed, exit code is 1, error msg is Check Failed: git revisions between shims are not equal
> ...
> Error: 6-13T22:10:21.501Z] [ERROR] Version file /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-4907/dist/target/parallel-world/spark330/rapids4spark-version-info.properties, revision is 65f2d79e1cc6fda420ffe5a0716da54d9b58c1d0
> Error: 6-13T22:10:21.501Z] [ERROR] Version file /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-4907/dist/target/parallel-world/spark340/rapids4spark-version-info.properties, revision is 165ec8127f829ede29c69cff729fcee82bcbd8e2
> ```

Yes, also nightly build would also be blocked by the issue. 
https://github.com/NVIDIA/spark-rapids/blob/branch-22.08/jenkins/version-def.sh#L52

There is an issue https://github.com/NVIDIA/spark-rapids/issues/5086 for tracking, it would nice to use maybe pom file to be the source of truth for all build/test scripts

**Comment by tgravescs** [comment]
filed https://github.com/NVIDIA/spark-rapids/issues/5823 to add separate nightly for 3.4.
filed https://github.com/NVIDIA/spark-rapids/issues/5824 to make sure we add 3.4 back into build in 22.10

**Comment by tgravescs** [comment]
build

**Review by abellina — APPROVED** [review]

**Pull request merged**